plugins/repository-azure/src/main/java/org/opensearch/repositories/azure/AzureStorageSettings.java

```diff
@@ -278,9 +296,39 @@ public String getConnectString() {
         return connectString;
     }
 
-    private static String buildConnectString(String account, @Nullable String key, @Nullable String sasToken, String endpointSuffix) {
+    public String getAccount() {
+        return account;
+    }
+
+    private static String validateTokenCredentialType(String tokenCredentialType) {
+        tokenCredentialType = tokenCredentialType.toUpperCase(Locale.ROOT);
+        final boolean hasTokenCredentialType = Strings.hasText(tokenCredentialType);
+        boolean isValidTokenCredentialType = false;
+        for (TokenCredentialType type : TokenCredentialType.values()) {
+            if (Objects.equals(type.toString(), tokenCredentialType)) {
+                isValidTokenCredentialType = true;
+            }
+        }
+        if (hasTokenCredentialType && !isValidTokenCredentialType) {
+            throw new SettingsException(String.format("'%s' is currently not supported.", tokenCredentialType));
+        }
+        return tokenCredentialType;
+    }
+
+    private static String buildConnectString(
+        String account,
+        @Nullable String key,
+        @Nullable String sasToken,
+        String endpointSuffix,
+        String tokenCredentialType
+    ) {
         final boolean hasSasToken = Strings.hasText(sasToken);
         final boolean hasKey = Strings.hasText(key);
+        final boolean hasTokenCredentialType = Strings.hasText(tokenCredentialType);
+        // When a valid token credential type is declared, we are no longer using connection string
+        if (hasTokenCredentialType) {
```

Comment:
Please remove that, the `buildConnectString` should never be used if managed credentials are configured

Reply:
fixed